### PR TITLE
Refactor: Changes time variables from int to chrono

### PR DESCRIPTION
**Pull request by shaavan**
This PR is a follow-up to #23801.
This PR aims to make the following changes to all the time variables in **net_processing.cpp** wherever possible.

- Convert all time variables to `std::chrono.`
- Use `chorno::literals` wherever possible.
- Use `auto` keywords wherever possible.
- Use `var{val}` convention of initialization.

This PR also minimizes the number of times, serialization of time `count_seconds(..)` occurs.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -57,14 +57,14 @@ static constexpr auto HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER = 1ms;
...
-static constexpr int64_t CHAIN_SYNC_TIMEOUT = 20 * 60; // 20 minutes
+/** Timeout for (unprotected) outbound peers to sync to our chainwork */
+static constexpr auto CHAIN_SYNC_TIMEOUT{20min};
 /** How frequently to check for stale tips */
 static constexpr auto STALE_CHECK_INTERVAL{10min};
 /** How frequently to check for extra outbound peers and disconnect */
 static constexpr auto EXTRA_PEER_CHECK_INTERVAL{45s};
 /** Minimum time an outbound-peer-eviction candidate must be connected for, in order to evict */
-static constexpr std::chrono::seconds MINIMUM_CONNECT_TIME{30};
+static constexpr auto MINIMUM_CONNECT_TIME{30s};
```
Comment (MarcoFalke) [inline src/net_processing.cpp:67]: This is already chrono. I think those changes should be in a separate commit (or left as is)

File: src/net_processing.cpp [hunk 2]
```diff
@@ -4348,7 +4348,7 @@ void PeerManagerImpl::CheckForStaleTipAndEvictPeers()
         // Check whether our tip is stale, and if so, allow using an extra
         // outbound peer
         if (!fImporting && !fReindex && m_connman.GetNetworkActive() && m_connman.GetUseAddrmanOutgoing() && TipMayBeStale()) {
-            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", count_seconds(now) - count_seconds(m_last_tip_update));
+            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", count_seconds(now - m_last_tip_update.load()));
```
Comment (MarcoFalke) [inline src/net_processing.cpp:4351]: nit: (Feel free to ignore). Maybe add a line break while touching this?

```suggestion
            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n",
count_seconds(now - m_last_tip_update.load()));
```

**Review by maflcko — COMMENTED** [review]
Copncept ACK. I haven't reviewed this, but I left comments

File: src/net_processing.cpp (hunk 1 above)
Comment (shaavan) [reply src/net_processing.cpp:67]: Let me post these changes as a separate commit.

File: src/net_processing.cpp [hunk 3]
```diff
@@ -88,19 +88,19 @@ static constexpr int32_t MAX_PEER_TX_REQUEST_IN_FLIGHT = 100;
...
+static constexpr auto TXID_RELAY_DELAY{2s};
 /** How long to delay requesting transactions from non-preferred peers */
-static constexpr auto NONPREF_PEER_TX_DELAY = std::chrono::seconds{2};
+static constexpr auto NONPREF_PEER_TX_DELAY{2s};
 /** How long to delay requesting transactions from overloaded peers (see MAX_PEER_TX_REQUEST_IN_FLIGHT). */
-static constexpr auto OVERLOADED_PEER_TX_DELAY = std::chrono::seconds{2};
+static constexpr auto OVERLOADED_PEER_TX_DELAY{2s};
 /** How long to wait (in microseconds) before downloading a transaction from an additional peer */
-static constexpr std::chrono::microseconds GETDATA_TX_INTERVAL{std::chrono::seconds{60}};
+static constexpr auto GETDATA_TX_INTERVAL{60us};
```
Comment (MarcoFalke) [inline src/net_processing.cpp:97]: This is wrong 60s is not equal to 60us

**Comment by shaavan** [comment]
Updated from 17af89fdaf0f95a21f36d54421d3bef745af13be to 16d0b619d72971b23eabcb0a5849370720f93909 ([pr23832.01](https://github.com/shaavan/bitcoin/tree/pr23832.01) -> [pr23832.02](https://github.com/shaavan/bitcoin/tree/pr23832.02))

Addressed @MarcoFalke comments

Changes:
- Split PR into two commits.
    - First one converting remaining time variables from **int** to **Chrono**
    - The second one makes the other changes done in this PR.
- Added line break between a long `LogPrintf` line.
- Corrected one error in variable initialization.

File: src/net_processing.cpp (hunk 3 above)
Comment (shaavan) [reply src/net_processing.cpp]: Sorry. I made an error interpreting this line of code earlier.

In the updated PR (commit 16d0b619d72971b23eabcb0a5849370720f93909), I have changed the variable to **std::chrono::seconds.**

`std::chrono::microseconds GETDATA_TX_INTERVAL{std::chrono::seconds{60}};` -> `auto GETDATA_TX_INTERVAL{60s};`

As I could not see any apparent reason to keep it in microseconds.

File: src/net_processing.cpp (hunk 2 above)
Comment (shaavan) [reply src/net_processing.cpp]: Addressed in e7535d44cbd4273283202d95c31dbd29e9945115

File: src/net_processing.cpp (hunk 1 above)
Comment (shaavan) [reply src/net_processing.cpp:67]: Addressed in 16d0b619d72971b23eabcb0a5849370720f93909

**Review by w0xlt — APPROVED** [review]
crACK 16d0b61

**Comment by klementtan** [comment]
crACK 16d0b619d72971b23eabcb0a5849370720f93909

* verified that the correct `std::chrono::*` is used when converting from `int` to `std::chrono`
* verified that the correct `chorno::literals` is used

**Review by klementtan — APPROVED** [review]

File: src/net_processing.cpp [hunk 4]
```diff
@@ -949,10 +949,10 @@ bool PeerManagerImpl::TipMayBeStale()
 {
     AssertLockHeld(cs_main);
     const Consensus::Params& consensusParams = m_chainparams.GetConsensus();
-    if (count_seconds(m_last_tip_update) == 0) {
+    if (m_last_tip_update.load() == 0s) {
```
Comment (MarcoFalke) [inline src/net_processing.cpp:952]: Wrong commit? This has nothing to do with switching types?

File: src/net_processing.cpp [hunk 5]
```diff
@@ -57,8 +57,8 @@ static constexpr auto HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER = 1ms;
  * behind headers chain.
  */
 static constexpr int32_t MAX_OUTBOUND_PEERS_TO_PROTECT_FROM_DISCONNECT = 4;
-/** Timeout for (unprotected) outbound peers to sync to our chainwork, in seconds */
-static constexpr int64_t CHAIN_SYNC_TIMEOUT = 20 * 60; // 20 minutes
+/** Timeout for (unprotected) outbound peers to sync to our chainwork */
```
Comment (MarcoFalke) [inline src/net_processing.cpp:60]: Will also need to remove "seconds" from where `CHAIN_SYNC_TIMEOUT` is used in docs?

**Review by maflcko — APPROVED** [review]
ACK 16d0b619d72971b23eabcb0a5849370720f9390 🔠

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 16d0b619d72971b23eabcb0a5849370720f9390 🔠
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgWGgv/XEpKkyorblBTjNK5oX47xGt3LfLxMXjnRW59LYfE9yo1W+TkYiEYALGN
ltJw9WQCyMpsJYLHrjL9ll6Z6w+ZE4F6MAZybTM2tOMzToDJgLT0+L6eQsb6HDgn
bZhp+HJXtWDrL+kD7RxpNbeeSDVRGDLTUElzuBQRacShUM+aXX2vrf6/tQyih1F+
H7E4EKH9f5ntWlxnT0l2h5ildkX6txh31fU3H9fwrKvRZ5bNZEogGF07fOaZFreE
y06UR4wxelPw4eRKdavhe79GlV6C+uFrf7SNLDzg7CziQTKJbU41m+WZ8f06gtQK
5EkT11i7tKaEiMbUeGIFDFOYsu9xCeTRr6oCM1nomb4iNREKAIClp24XtiN8hQye
kN7QPWoPot5LOk6geT5UN0mti6j5O//fW8kT+7Z1D7aFI7fSBwVqHNjK+YAnXrrk
O3TSZVyXu/+ljzvYWLZUPpqQaa3durNCjOFtV+VyysP7ApasVXK6W/FXx9Zf2uiw
49855sGB
=q2XZ
-----END PGP SIGNATURE-----
```


</details>

File: src/net_processing.cpp (hunk 5 above)
Comment (shaavan) [reply src/net_processing.cpp:60]: Just to confirm. Are you talking about this line?

https://github.com/bitcoin/bitcoin/blob/d69af93223c4008c3255f7e4848ff05d78c514fa/src/net_processing.cpp#L729-L730

Comment (MarcoFalke) [reply src/net_processing.cpp:60]: Yes

Comment (shaavan) [reply src/net_processing.cpp:60]: This line could use a fixing.

However, doing so will revoke the current 3 ACKs on the current top commit of this PR.
So would you suggest I should update this PR for this change?

Comment (MarcoFalke) [reply src/net_processing.cpp:60]: If it is fixed, it should be fixed here to avoid another follow-up pull. If not, it might be best to leave it as-is.

A re-ACK is less overhead and effort than a new pull + fresh ACK.

**Comment by shaavan** [comment]
Updated from 16d0b619d72971b23eabcb0a5849370720f93909 to fe86eb50c986f7b5ccce63f984d8a51cd9ee9e2c ([pr23832.02](https://github.com/shaavan/bitcoin/tree/pr23832.02) -> [pr23832.03](https://github.com/shaavan/bitcoin/tree/pr23832.03), [diff](https://github.com/shaavan/bitcoin/compare/pr23832.02..pr23832.03))
Addressed @MarcoFalke suggestion.

- Removed the term _second_ for **CHAIN_SYNC_TIMEOUT** in docs.

Should be trivial to reACK.

File: src/net_processing.cpp (hunk 5 above)
Comment (shaavan) [reply src/net_processing.cpp:60]: Fixed in fe86eb50c986f7b5ccce63f984d8a51cd9ee9e2c

**Review by maflcko — APPROVED** [review]
re-ACK fe86eb50c986f7b5ccce63f984d8a51cd9ee9e2c  🏕

only minor doc change

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK fe86eb50c986f7b5ccce63f984d8a51cd9ee9e2c  🏕

only minor doc change
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhSQAv/aT3khOx+9Ns7XrLCiJPyXT33dwP69vqblTWRrSrga/mRkuvsGrAG89xG
z3zhR8Mc5KUA2+OOshFwM4twxSzh/8ugcWAiIQ8P0zkmF+oL+wV1K0l3TLf8W8Vt
gPhIgKt+7t+lNZYvvMCD3bTImrG0/0+/TpuV6NaP+hNKGEmswOaXoqle/kez2ZXU
ObsjBJvlSk3EeBZRDQxh3ZuO99tkhHFM4OzHaAdhoE8zi0c6FCi+uNxSDPs/UH6Z
rYIJnSAYxUZ/DnJUqLZTd59N/tUAZPjBbs/yso6V/tr7yt4tbVVaPaOSzbj/e95B
pqg3xSssL6KYqUzeYudFJY0K2S9nvQM+si7STkqFZ3aSPf1GRcXCbkUEoqVENRMG
yrgIaSC23WrRMas+CSk5gaqKHUTlpM/6WV8KnNaosQbSBhU7mi/lTf6SZNc5XomC
mkek8R4TRft1Q7qivmOrVsSlBuGrrCRjMtc7ddwlJ2f2bz2kLgkwUzw1hsGcf+Lj
nmp944H6
=S8MN
-----END PGP SIGNATURE-----
```


</details>
